rust_snuba/src/consumer.rs

```diff
@@ -67,25 +75,79 @@ pub fn consumer_impl(
                 acc
             });
 
-            let transform_step = PythonTransformStep::new(
-                self.processor_config.clone(),
-                self.processes,
-                Reduce::new(
-                    Box::new(ClickhouseWriterStep::new(
-                        CommitOffsets::new(Duration::from_secs(1)),
-                        self.clickhouse_cluster_config.clone(),
-                        self.clickhouse_table_name.clone(),
-                        self.skip_write,
-                        2,
-                    )),
-                    accumulator,
-                    BytesInsertBatch { rows: vec![] },
-                    self.max_batch_size,
-                    self.max_batch_time,
+            let next_step = Reduce::new(
+                Box::new(ClickhouseWriterStep::new(
+                    CommitOffsets::new(Duration::from_secs(1)),
+                    self.clickhouse_cluster_config.clone(),
+                    self.clickhouse_table_name.clone(),
+                    self.skip_write,
+                    2,
+                )),
+                accumulator,
+                BytesInsertBatch { rows: vec![] },
+                self.max_batch_size,
+                self.max_batch_time,
+            );
+
+            match (
+                self.use_rust_processor,
+                processors::get_processing_function(&self.processor_config.python_class_name),
+            ) {
+                (true, Some(func)) => {
+                    struct MessageProcessor {
+                        func: fn(
```

Comment:
I think you can write a `impl<T> TaskRunner for T where T: FnMut` inside of `run_task_in_threads`, so you can use the `fn` directly as a TaskRunner. Or alternatively I think it'd be best if the MessageProcessor here would be a general-purpose utility.